docs/understand/programming_model.rst

```diff
@@ -0,0 +1,158 @@
+vector Arithmetic Logical Unit (ALU). Most recent GPUs beside the vector ALU
+also house some matrix ALU for accelerating algorithms of well defined shapes.
+
+Some number of vector and matrix ALUs comprise a larger block, often referred
+to as a Compute Unit (OpenCL, AMD block diagrams) but is referred to as Multi
+Processor in HIP terms.
+
+.. _rdna3_cu:
+
+.. figure:: ../data/understand/programming_model/rdna3_cu.png
+  :alt: Block diagram showing components mostly duplicated on the upper and
+        lower halves of the image with some spanning over both parts (hinting
+        at them being shared). Both the top and the bottom have two sets of
+        identical hardware blocks. a "Scheduler" having "Vector GPR"s with an
+        associated Vector ALU with the following noted capabilities:
+        Float/INT/Matrix SIMD32, Float/Matrix SIMD32, Transcendental SIMD8, AI
+        MATRIX Accelerator, DPFP (1). Both top and bottom have a single "Ray
+        Accelerator", "Texture Filters" and "LD/ST/Tex Addr" and L0 blocks.
+        Shared among top and bottom are "Scalar Cache", "Shader Instruction
+        Cache" and Shared Memory.
+
+  Block Diagram of an RDNA3 Compute Unit.
+
+.. _cdna3_cu:
+
+.. figure:: ../data/understand/programming_model/cdna3_cu.png
+  :alt: Block diagram showing components: "Scheduler", "Local Data Share",
+        "Matrix Core Unit", "Shader Core" and "L1 Cache" along with some
+        unnamed blocks.
+
+  Block Diagram of an CDNA3 Compute Unit.
+
+For hardware implementation's sake, some number of Multi Processors are grouped
+together into a Shader Engine or Compute Engine, typically sharing some fixed
+function units or memory subsystem resources.
+
+.. _cdna2_gcd:
+
+.. figure:: ../data/understand/programming_model/cdna2_gcd.png
+  :alt: Block diagram showing four "Compute Engine"s each with 28 "CU"s
+        (Compute Unit) inside. These four Compute Engines share one block of
+        "L2 Cache and Controllers". Around them are four "Memory Controller"s,
+        each having a "Memory Phy" block next to them. To the top and bottom of
+        all these are eight blocks of "Infinity Fabric Link" with one of the
+        eight reading "Infinity Fabric or PCIe". Two sole "VCN" blocks sit in
+        top corners. At the very bottom spans a colored seaction reading
+        "Infinity Fabric" along with its logo.
+
+  Block Diagram of a CDNA2 Graphics Compute Die.
+
+Single Instruction Multiple Threads
+===============================================================================
+
+The SIMT programming model behind the HIP device-side execution is a
+middle-ground between SMT (Simultaneous Multi-Threading) programming known from
+multi-core CPUs, and SIMD (Single Instruction, Multiple Data) programming
+mostly known from exploiting relevant instruction sets on CPUs (eg.
+SSE/AVX/Neon).
+
+A HIP device compiler maps our SIMT code written in HIP C++ to an inherently
```

Comment:
This is a change of tone from the previous sections. "the" could also be ommited
```suggestion
A HIP device compiler maps (the) SIMT code written in HIP C++ to an inherently
```

Reply:
Fixed